lib/mix/lib/mix/dep/lock.ex

```diff
@@ -62,4 +63,11 @@ defmodule Mix.Dep.Lock do
   defp lockfile do
     Mix.Project.config[:lockfile]
   end
+
+  defp assert_no_merge_conflicts_in_lockfile(lockfile, info) do
+    if info =~ ~r/^(?:<<<<<<<|=======|>>>>>>>|\|\|\|\|\|\|\|)/ do
```

Comment:
This is what we discussed here: https://github.com/elixir-lang/elixir/pull/4482#discussion_r58687657
With `String.contains?` it is harder to match on the beginning of the line.
What do you think?